app/controllers/api/v1/users_controller.rb

```diff
@@ -27,6 +27,19 @@ def show
       end
     end
 
+    def search
+      if params[:q].present?
+        users = User.where("name ~* ?", params[:q])
+      else
+        users = User.none
+      end
+      users = users.by_email
```

Comment:
How am I supposed to test it then if the order of the same-named results isn't well defined? :disappointed: